execute/plugin.go

```diff
@@ -205,32 +213,57 @@ func getPendingExecutedReports(
 
 		ranges, err := computeRanges(reports)
 		if err != nil {
-			return nil, nil, fmt.Errorf("compute report ranges: %w", err)
+			return nil, nil, nil, fmt.Errorf("compute report ranges: %w", err)
 		}
 
-		executedMessageSet := mapset.NewSet[cciptypes.SeqNum]()
+		// Get both finalized and unfinalized executed messages
+		finalizedMsgSet := mapset.NewSet[cciptypes.SeqNum]()
+		allExecutedMsgSet := mapset.NewSet[cciptypes.SeqNum]()
+
 		for _, seqRange := range ranges {
-			executedMessagesForRange, err2 := ccipReader.ExecutedMessages(ctx, selector, seqRange)
-			if err2 != nil {
-				return nil, nil, fmt.Errorf("get %d executed messages in range %v: %w", selector, seqRange, err2)
+			// Get all executed messages
+			allMessages, err := ccipReader.ExecutedMessages(ctx, selector, seqRange, primitives.Unconfirmed)
+			if err != nil {
+				return nil, nil, nil, fmt.Errorf("get %d executed messages in range %v: %w", selector, seqRange, err)
+			}
+			allExecutedMsgSet = allExecutedMsgSet.Union(mapset.NewSet(allMessages...))
+
+			// Get finalized messages
+			finalizedMessages, err := ccipReader.ExecutedMessages(ctx, selector, seqRange, primitives.Finalized)
+			if err != nil {
+				return nil, nil, nil, fmt.Errorf("get finalized %d executed messages in range %v: %w", selector, seqRange, err)
 			}
-			executedMessageSet = executedMessageSet.Union(mapset.NewSet(executedMessagesForRange...))
+			finalizedMsgSet = finalizedMsgSet.Union(mapset.NewSet(finalizedMessages...))
 		}
 
-		executedMessages := executedMessageSet.ToSlice()
-		sort.Slice(executedMessages, func(i, j int) bool { return executedMessages[i] < executedMessages[j] })
+		// Get unfinalized messages by taking the difference
+		unfinalizedMsgSet := allExecutedMsgSet.Difference(finalizedMsgSet)
 
-		// Remove fully executed reports.
-		// Populate executed messages on the reports.
-		var executedCommits []exectypes.CommitData
-		groupedCommits[selector], executedCommits = combineReportsAndMessages(reports, executedMessages)
-		fullyExecuted = append(fullyExecuted, executedCommits...)
+		finalizedMessages := finalizedMsgSet.ToSlice()
```

Comment:
nit, but this could be extracted to a seperate helper function
```go

function toSortedSlice(set mapset.Set) []whatever {
                messages := set.ToSlice()
		sort.Slice(messages, func(i, j int) bool { return messages[i] < messages[j] })
                 return messages
}
```